xdsl/dialects/dlti.py

```diff
@@ -89,6 +130,12 @@ class DataLayoutSpecAttr(DLTIEntryMap):
 
     name = "dlti.dl_spec"
 
+    def __init__(
+        self,
+        contents: ArrayAttr[DataLayoutEntryAttr] | DictValueType,
+    ):
+        return super().__init__(contents)
```

Comment:
Does something break if this is not added? I would expect things to work without this override

Reply:
Can you try adding `@datclass(frozen=True)` on `DLTIEntryMap`?

Reply:
I don't think this works with Pyright, as the error `Argument of type "Mapping[StringAttr | TypeAttribute | str, Attribute | str | int | DictValueType]" cannot be assigned to parameter "entries" of type "ArrayAttr[DataLayoutEntryAttr]" in function "__init__"`

I think this is because the generated __init__ for these subclasses is typed just `ArrayAttr[DataLayoutEntryAttr]` which follows the type of `entries` variable, whereas we want `ArrayAttr[DataLayoutEntryAttr] | DictValueType` for the argument  which is then passed to the super class __init__ method for handling

Reply:
Yer, I tried init=False but that had the same issue - it feels like it should work and seems a bit nasty having to duplicate the init method

Reply:
Ah cool, applied to the code :)